pkg/fanal/analyzer/language/python/poetry/poetry.go

```diff
@@ -105,34 +105,75 @@ func (a poetryAnalyzer) mergePyProject(fsys fs.FS, dir string, app *types.Applic
 
 	// Identify the direct/transitive dependencies
 	for i, pkg := range app.Packages {
-		if _, ok := p[pkg.Name]; ok {
+		if _, ok := project.Tool.Poetry.Dependencies[pkg.Name]; ok {
 			app.Packages[i].Relationship = types.RelationshipDirect
 		} else {
 			app.Packages[i].Indirect = true
 			app.Packages[i].Relationship = types.RelationshipIndirect
 		}
 	}
 
+	prodDeps := getProdDeps(project, app)
+
+	app.Packages = lo.Filter(app.Packages, func(pkg types.Package, _ int) bool {
+		_, ok := prodDeps[pkg.ID]
+		return ok
+	})
```

Comment:
I am thinking about moving logic into `getProdDeps` and renaming  function to `filterDevDeps`.
wdyt?

Reply:
OK, but I think it's more correct to call it `filterProdPackages`, since `filderDev` stands for dependency filtering and dev extraction.